### PR TITLE
chore: switch `new syntax` issue template to forms

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[X] Other, please explain:

#### What changes did you make? (Give an overview)

switch `new syntax` issue template to forms.

[Quick Preview Link](https://github.com/eslint/eslint/blob/docs/new-syntax-template/.github/ISSUE_TEMPLATE/new-syntax.yml)

I kept the `Implementation Checklist` part in the markdown because of the sub-tasks for pasting links.

#### Is there anything you'd like reviewers to focus on?

No

### CODE REVIEW

**Review by aladdin-add — APPROVED** [review]
LGTM! 💯

File: .github/ISSUE_TEMPLATE/new-syntax.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  validations:
+    required: true
+- type: textarea
+  attributes:
+    label: Example code
+    description: Please provide some example code for the new syntax.
+    render: js
+  validations:
+    required: true
+- type: textarea
```
Comment (nzakas) [inline .github/ISSUE_TEMPLATE/new-syntax.yml:31]: This should also be done as text boxes to render properly. The URLs will just need to be added after the issue is created.

**Review by nzakas — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/new-syntax.yml (hunk 1 above)
Comment (snitin315) [reply .github/ISSUE_TEMPLATE/new-syntax.yml]: Resolved.

**Review by nzakas — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**